pkg/errors/errors_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package errors_test
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/mainflux/mainflux/pkg/errors"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	msg1  = "message 1"
+	msg2  = "message 2"
+	msg3  = "message 3"
+	level = 10
+)
+
+var (
+	err1 = errors.New(msg1)
+	err2 = errors.New(msg2)
+	err3 = errors.New(msg3)
+)
+
+type customError struct {
+	msg string
+}
+
+func (ce *customError) Error() string {
+	return ce.msg
+}
+
+func TestError(t *testing.T) {
+	cases := []struct {
+		desc string
+		err  error
+		msg  string
+	}{
+		{
+			desc: "level 0 wrapped error",
+			err:  err1,
+			msg:  msg1,
+		},
+		{
+			desc: "level 1 wrapped error",
+			err:  errors.Wrap(err1, err2),
+			msg:  fmt.Sprintf("%s : %s", msg1, msg2),
+		},
+		{
+			desc: "level 2 wrapped error",
+			err:  errors.Wrap(err1, errors.Wrap(err2, err3)),
+			msg:  fmt.Sprintf("%s : %s : %s", msg1, msg2, msg3),
+		},
+		{
+			desc: fmt.Sprintf("level %d wrapped error", level),
+			err:  wrap(level),
+			msg:  message(level),
```

Comment:
Why you are not using error messages here?

Reply:
Function `message` produces error message of n level wrapped message. We use this error message to compare it with the error of an actually wrapped error with func `err`.

Reply:
What I mean is that for other tests you have an error message. You could have a funcrtion `createWrap(msg, level)` that you could use to generate wrapped error in all your tests. For example, you could replace `fmt.Sprintf("%s : %s : %s", msg1, msg2, msg3)`  by `createWrap(msg, 3)`

Reply:
Yeah, maybe a unified func would be better. For now, I've replaced `fmt.Sprintf("%s : %s : %s", msg1, msg2, msg3)` with `message` func.